### PR TITLE
Fix rare bug in config loading + add unit test

### PR DESCRIPTION
**Pull request by jdavcs**
The value of a path segment may be None if the config option is set to None in kwargs.
This may be a rare scenario, but it may happen, and it does happen when
running integration tests: job_config_file is overridden with `None`.

PR includes fix + new unit test that exposes the bug.

### CODE REVIEW

File: lib/galaxy/config/__init__.py [hunk 1]
```diff
@@ -283,7 +283,8 @@ def strip_deprecated_dir(key, value):
         for key, value in kwargs.items():
             if key in self.appschema:
                 value = convert_datatype(key, value)
-                value = strip_deprecated_dir(key, value)
+                if value:
```
Comment (mvdbeek) [inline lib/galaxy/config/__init__.py:286]:
```suggestion
                if value is not None:
```
Based on the description that seems more correct to me ?

Comment (jdavcs) [reply lib/galaxy/config/__init__.py:286]: It should be `if value` (or `if value is not None`) . Sorry - my description does sound confusing: I meant that it *may* be None, in which case the code breaks, so this is a fix (because the code will try to split a None value). 
~~That said, while looking into this I found a case that would break things: value may be not a string (if set by mistake).~~ [EDIT: checked, not possible, so all good.] .  I'll test for more edge cases and will submit a fix later today.

Comment (jdavcs) [reply lib/galaxy/config/__init__.py:286]: I checked. In this case it cannot be a wrong datatype: so no additional bug. 
May I mark as resolved?
